extensions/discovery-and-network/services.yaml

```diff
@@ -0,0 +1,247 @@
+        source: >-
+          $count(status.loadBalancer.ingress) ? $map(status.loadBalancer.ingress,
+          function($value) {$value.ip ? $value.ip : $value.hostname}) ~> $join(',
+          '): ($count($item.spec.externalIPs) ? $item.spec.externalIPs ~> $join(',
+          '): '-')
+    body:
+      - widget: ResourceList
+        source: $subscriptions()
+        name: Subscription
+        children:
+          - source: metadata.name
+            name: Name
+            sort: 'true'
+            widget: ResourceLink
+            resource:
+              name: metadata.name
+              namespace: $root.metadata.namespace
+              kind: kind
+          - source: spec.filter.filters[0].eventType.value
+            name: First event type
+            sort:
+              default: true
+              compareFunction: $compareStrings($first, $second)
+          - source: status.ready
+            name: Ready
+            sort: true
+            widget: Badge
+            highlights:
+              positive:
+                - 'true'
+              negative:
+                - 'false'
+      - widget: ResourceList
+        source: $apirules()
+        name: API Rule
+        sort:
+          - source: spec.service.host
+            default: true
+          - source: spec.service.port
+            compareFunction: $first - $second
+  form: |-
+    - widget: KeyValuePair
+      path: spec.selector
+      simple: true
+      name: Selectors
+      placeholder: spec.selector
+    - path: spec.type
+      name: Type
+      placeholder: Enter Type
+    - widget: GenericList
+      path: spec.ports
+      simple: true
+      children:
+        - path: '[]'
+          simple: true
+          children:
+            - path: name
+              simple: true
+              name: Ports Name
+              placeholder: Enter Ports Name
+            - path: protocol
+              simple: true
+              name: Protocol
+              placeholder: Enter Protocol
+              widget: Text
+              required: true
+            - path: port
+              simple: true
+              name: Port
+              placeholder: Enter Port
+              required: true
+            - path: targetPort
+              simple: true
+              name:  Target Port
+              placeholder: Enter Target Port
+              widget: Text
+              required: true
+              inputInfo: Number or name of the port to access on the pods targeted by the service. Number must be in the range 1 to 65535.
+            - path: nodePort
+              simple: true
+              name: Node Port
+              placeholder: Enter Node Port
+            - path: appProtocol
+              simple: true
+              name: Application Protocol
+              placeholder: Enter Application Protocol
+      template:
+        protocol: 'TCP'
+        port: 80
+        targetPort: 9376 
+    - widget: SimpleList
+      path: spec.ipFamilies
```

Comment:
this could be an enum 
https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services